### ISSUE TITLE
Difference API results and local diffuser results

### ISSUE THREAD

**Issue by CaffreyR** [huggingface/diffusers#6357]
### Describe the bug

Hi, I train a diffuser model and push to the hub. And I find that I cannot reproduce the result of the hub api using codes like this and it seems the results in the API are better.

```
from diffusers import StableDiffusionPipeline, EulerDiscreteScheduler
import torch
model_id = "my_repo"
scheduler = EulerDiscreteScheduler.from_pretrained(model_id, subfolder="scheduler")
pipe = StableDiffusionPipeline.from_pretrained(model_id, scheduler=scheduler, torch_dtype=torch.float16)
pipe = pipe.to("cuda")

prompt = "A photo of the sks face of the nurse"
image = pipe(prompt).images[0]
image.save("generated_image.png")
```

Thanks a lot!

### Reproduction

from diffusers import StableDiffusionPipeline, EulerDiscreteScheduler
import torch
model_id = "my_repo"
scheduler = EulerDiscreteScheduler.from_pretrained(model_id, subfolder="scheduler")
pipe = StableDiffusionPipeline.from_pretrained(model_id, scheduler=scheduler, torch_dtype=torch.float16)
pipe = pipe.to("cuda")

prompt = "A photo of the sks face of the nurse"
image = pipe(prompt).images[0]
image.save("generated_image.png")

### System Info

- `diffusers` version: 0.25.0.dev0
- Platform: Linux-6.2.0-37-generic-x86_64-with-glibc2.35
- Python version: 3.9.18
- PyTorch version (GPU?): 2.1.2+cu121 (True)
- Huggingface_hub version: 0.19.4
- Transformers version: 4.36.1
- Accelerate version: 0.25.0
- xFormers version: not installed
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help?

@yiyixuxu @DN6 @sayakpaul @patrickvonplaten

**Comment by sayakpaul** [comment]
You need to use a fixed seed and also ensure the environments are the same to some extent. The `num_inference_steps`, `guidance_scale`, etc. parameters need to be fiddled around with, too. Long story cut short, you need to experiment to be able to reproduce.

**Comment by CaffreyR** [comment]
Hi @sayakpaul , so how can I get the parameter of the API of Huggingface Hub

**Comment by sayakpaul** [comment]
Cc: @oOraph regarding that.

**Comment by DN6** [comment]
@CaffreyR I'm not sure I understand what you mean by Hub API here? Could you please provide a snippet that demonstrates the difference in quality?

**Comment by sayakpaul** [comment]
They probably meant the inference API that is used under the hood when you click on the inference widget of text-to-image repository on the Hub.

**Comment by CaffreyR** [comment]
Yes, that's it. I train a dreambooth but it turns differently when I use the API and local model. @DN6 @sayakpaul 
<img width="647" alt="image" src="https://github.com/huggingface/diffusers/assets/84232793/cd555946-59b8-428c-ad4f-ab7405b1dc81">
<img width="506" alt="image" src="https://github.com/huggingface/diffusers/assets/84232793/1b7c88c0-6c48-4b0d-8c58-d9e5a2792511">

**Comment by DN6** [comment]
Hi @CaffreyR is the issue here that the the inference API and your code snippet are not producing the same image? If so, that is probably because the inference API does not use fixed a seed before running the generation process.

**Comment by CaffreyR** [comment]
Dear @DN6 , the issue is API seems to produce better and fine-grained images. And how can I get the API parameter and seeds? Thanks

**Comment by DN6** [comment]
@CaffreyR The inference API will use the default pipeline parameters. You can try

```python
from diffusers import StableDiffusionPipeline
import torch

model_id = "my_repo"
pipe = StableDiffusionPipeline.from_pretrained(model_id)
pipe = pipe.to("cuda")

prompt = "A photo of the sks face of the nurse"
image = pipe(prompt).images[0]
image.save("generated_image.png")
```

**Comment by sayakpaul** [comment]
Going to close this for inactivity. Feel free to re-open.
